pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -723,7 +723,10 @@ private void setupSerDe(Class<?>[] typeArgs, ClassLoader clsLoader) {
         this.inputSerDe = new HashMap<>();
         instanceConfig.getFunctionConfig().getCustomSerdeInputsMap().forEach((k, v) -> this.inputSerDe.put(k, initializeSerDe(v, clsLoader, typeArgs, true)));
         for (String topicName : instanceConfig.getFunctionConfig().getInputsList()) {
-            this.inputSerDe.put(topicName, initializeDefaultSerDe(typeArgs, true));
+            // initialize Default-SerDe if custom-SerDe is not attached to the topic
```

Comment:
Isn't a better check to verify that the same topic is not in both inputs and customSerdeInput?

Reply:
> Isn't a better check to verify that the same topic is not in both inputs and customSerdeInput?

Please correct if I am wrong. But inputs is filled by CustomSerde-map only at line#724. So, if CustomSerDe is provided and if we still try to init with DefaultSerDe then we see below exception. 

```
Caused by: java.lang.RuntimeException: Default Serializer does not support class org.apache.pulsar.functions.instance.JavaInstanceRunnableTest$TestData
	at org.apache.pulsar.functions.instance.JavaInstanceRunnable.initializeDefaultSerDe(JavaInstanceRunnable.java:711)
	at org.apache.pulsar.functions.instance.JavaInstanceRunnable.setupSerDe(JavaInstanceRunnable.java:729)
```

Reply:
Line 724 is going over the customSerDeInput List and filling up the inputSerDe map, while line 726 goes over input List to fill the inputSerDe map. There are three cases
1) only inputs with custom serde are specified. In which case 726 is a noop
2) only inputs with default serde are specified. In which case 724 is noop
3) Function takes some inputs specified by custom serdes and one inputs using defaultserde. In this case we need to populate the inputSerDe for those topics which are going to be read using the default serializer.